### PR TITLE
File-based table provider for Datafusion

### PR DESCRIPTION
**Pull request by AdamGS**
This PR is a very minimal `TableProvider` to read vortex-encoded files.

### CODE REVIEW

File: vortex-serde/src/file/reader/filtering.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@ use vortex_error::VortexResult;
 
 use super::projections::Projection;
 
-pub trait FilteringPredicate {
+pub trait FilteringPredicate: Send + Sync {
```
Comment (AdamGS) [inline vortex-serde/src/file/reader/filtering.rs:6]: ~~Datafusion basically requires it~~

Comment (AdamGS) [reply vortex-serde/src/file/reader/filtering.rs:6]: Should be `Send + 'static`, I'll fix that in a separate PR as it requires some unwinding around the IO code

File: vortex-datafusion/Cargo.toml [hunk 2]
```diff
@@ -30,10 +36,16 @@ datafusion-physical-plan = { workspace = true }
...
+log = { workspace = true }
+object_store = { workspace = true }
 pin-project = { workspace = true }
 
+
 [dev-dependencies]
-tokio = { workspace = true, features = ["test-util"] }
+anyhow = { workspace = true }
+url = "2"
+tempfile = "3"
```
Comment (robert3005) [inline vortex-datafusion/Cargo.toml]: move this to workspace

File: vortex-datafusion/src/persistent/opener.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        if let Some(_predicate) = self.predicate.as_ref() {
+            log::warn!("Missing logic to turn a physical expression into a RowFilter");
+        }
+
+        if let Some(projection) = self.projection.as_ref() {
+            builder = builder.with_projection(Projection::new(projection))
+        }
+
+        DFResult::Ok(
```
Comment (robert3005) [inline vortex-datafusion/src/persistent/opener.rs]: We should figure out fmt/clippy rule for this

Comment (robert3005) [reply vortex-datafusion/src/persistent/opener.rs]: There's a rule, however, it's very fiddly to configure. Will have a look at some point if there's a simple setup

File: vortex-serde/src/file/reader/filtering.rs (hunk 1 above)
Comment (robert3005) [reply vortex-serde/src/file/reader/filtering.rs:6]: This will change in my pr. The api here doesn't work for our serialization format.

**Review by robert3005 — APPROVED** [review]

**Comment by robert3005** [comment]
I wonder how datafusion handles massive file listing. I have a feeling it might just oom

**Pull request merged**